x-pack/legacy/plugins/maps/public/layers/layer.js

```diff
@@ -52,8 +52,8 @@ export class AbstractLayer {
   }
 
   destroy() {
-    if (this._source) {
-      this._source.destroy();
+    if (this.getSource()) {
```

Comment:
tbh. I wouldn't call getSource() here. it obfuscated more than it illuminates. This is overridden in blended_vector_layer anyway and is very readable there.